core/src/test/scala/org/apache/spark/AccumulatorSuite.scala

```diff
@@ -135,5 +137,22 @@ class AccumulatorSuite extends FunSuite with Matchers with LocalSparkContext {
       resetSparkContext()
     }
   }
+  
+  test ("garbage collection") {
+    // Create an accumulator and let it go out of scope to test that it's properly garbage collected
+    sc = new SparkContext("local", "test")
+    var acc: Accumulable[mutable.Set[Any], Any] = sc.accumulable(new mutable.HashSet[Any]())
+    val accId = acc.id
+    val ref = WeakReference(acc)
+
+    // Ensure the accumulator is present
+    assert(ref.get.isDefined)
+
+    // Remove the explicit reference to it and allow weak reference to get garbage collected
+    acc = null
+    System.gc()
+    assert(ref.get.isEmpty)
+    assert(Accumulators.originals.get(accId).isDefined)
```

Comment:
Hi Josh, how do I register a cleanup task ? I see a private method within ContextCleaner: registerForCleanup , that it seems I can use but I don't have visibility of that method within the Accumulator class. How would you recommend doing this?